server/views/pages/overview/partials/overviewTabContents.njk

```diff
@@ -9,11 +9,13 @@
         <strong>Goals</strong>
       </div>
       <div>
-        {{ govukButton({
-          text: "Add a new goal",
-          href: "/plan/" + prisonNumber + "/goals/create",
-          id: "add-goal-button"
-        }) }}
+        {% if hasEditAuthority %}
```

Comment:
The variable `hasEditAuthority` is a variable setup on response locals in `populateUserAuthorities` above